File: Serenity Practice/HosbitalSerenity.Web/Modules/Hosbital/DoctorSpecialities/DoctorSpecialitiesColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace HosbitalSerenity.Hosbital.Columns;

[ColumnsScript("Hosbital.DoctorSpecialities")]
[BasedOnRow(typeof(DoctorSpecialitiesRow), CheckNames = true)]
public class DoctorSpecialitiesColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int DoctorSpecialityId { get; set; }
    public string DoctorName { get; set; }
    public string SpecialityName { get; set; }
}

File: Serenity Practice/HosbitalSerenity.Web/Modules/ServerTypes/Hosbital/DoctorSpecialitiesColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { DoctorSpecialitiesRow } from "./DoctorSpecialitiesRow";

export interface DoctorSpecialitiesColumns {
    DoctorSpecialityId: Column<DoctorSpecialitiesRow>;
    DoctorName: Column<DoctorSpecialitiesRow>;
    SpecialityName: Column<DoctorSpecialitiesRow>;
}

export class DoctorSpecialitiesColumns extends ColumnsBase<DoctorSpecialitiesRow> {
    static readonly columnsKey = 'Hosbital.DoctorSpecialities';
    static readonly Fields = fieldsProxy<DoctorSpecialitiesColumns>();
}